service/rtc/server.go

```diff
@@ -94,77 +87,45 @@ func (s *Server) Start() error {
-					err := unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEADDR, 1)
-					if err != nil {
-						s.log.Error("failed to set reuseaddr option", mlog.Err(err))
-						return
-					}
-					err = unix.SetsockoptInt(int(fd), unix.SOL_SOCKET, unix.SO_REUSEPORT, 1)
-					if err != nil {
-						s.log.Error("failed to set reuseport option", mlog.Err(err))
-						return
-					}
-				})
-			},
-		}
-
-		listenAddress := fmt.Sprintf("%s:%d", s.cfg.ICEAddressUDP, s.cfg.ICEPortUDP)
-		udpConn, err := listenConfig.ListenPacket(context.Background(), "udp4", listenAddress)
+	var err error
+	var ips []string
+	var muxes []ice.UDPMux
+	if s.cfg.ICEAddressUDP == "" {
+		s.log.Debug("no address specified, going to listen on all supported interfaces")
+		ips, err = getSystemIPs(s.log)
 		if err != nil {
-			return fmt.Errorf("failed to listen on udp: %w", err)
-		}
-
-		s.log.Info(fmt.Sprintf("rtc: server is listening on udp %s", listenAddress))
-
-		if err := udpConn.(*net.UDPConn).SetWriteBuffer(udpSocketBufferSize); err != nil {
-			s.log.Warn("rtc: failed to set udp send buffer", mlog.Err(err))
+			return fmt.Errorf("failed to get system IPs: %w", err)
 		}
-
-		if err := udpConn.(*net.UDPConn).SetReadBuffer(udpSocketBufferSize); err != nil {
-			s.log.Warn("rtc: failed to set udp receive buffer", mlog.Err(err))
+		if len(ips) == 0 {
+			return fmt.Errorf("no valid address to listen on was found")
 		}
+	} else {
+		ips = append(ips, s.cfg.ICEAddressUDP)
+	}
 
-		connFile, err := udpConn.(*net.UDPConn).File()
+	for _, ip := range ips {
+		listenAddress := fmt.Sprintf("%s:%d", ip, s.cfg.ICEPortUDP)
+		conns, err := createUDPConnsForAddr(s.log, listenAddress)
 		if err != nil {
-			return fmt.Errorf("failed to get udp conn file: %w", err)
+			return fmt.Errorf("failed to create UDP connections: %w", err)
 		}
-		defer connFile.Close()
 
-		sysConn, err := connFile.SyscallConn()
+		udpConn, err := newMultiConn(conns)
 		if err != nil {
-			return fmt.Errorf("failed to get syscall conn: %w", err)
-		}
-		err = sysConn.Control(func(fd uintptr) {
-			writeBufSize, err := syscall.GetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_SNDBUF)
-			if err != nil {
-				s.log.Error("failed to get buffer size", mlog.Err(err))
-				return
-			}
-			readBufSize, err := syscall.GetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_RCVBUF)
-			if err != nil {
-				s.log.Error("failed to get buffer size", mlog.Err(err))
-				return
-			}
-			s.log.Debug("rtc: udp buffers", mlog.Int("writeBufSize", writeBufSize), mlog.Int("readBufSize", readBufSize))
-		})
-		if err != nil {
-			return fmt.Errorf("Control call failed: %w", err)
+			return fmt.Errorf("failed to create multiconn: %w", err)
 		}
 
-		conns = append(conns, udpConn)
-	}
-	var err error
-	s.udpConn, err = newMultiConn(conns)
-	if err != nil {
-		return fmt.Errorf("failed to create multiconn: %w", err)
+		muxes = append(muxes, ice.NewUDPMuxDefault(ice.UDPMuxParams{
+			Logger:  newPionLeveledLogger(s.log),
```

Comment:
Turns out there was a warning printed but we didn't set any logger:

```
ice WARNING: 2023/02/14 18:19:24 UDPMuxDefault should not listening on unspecified address, use NewMultiUDPMuxFromPort instead
```

It's unlikely we would have caught this in the e2e tests pipeline but still something. Now passing a wrapper to our logger in order to potentially catch these things sooner in the future.